src/javascript/tests.rs

```diff
@@ -275,3 +275,40 @@ function $isObject(object) {
 "#
     );
 }
+
+#[test]
+// https://xjamundx.medium.com/custom-javascript-errors-in-es6-aa891b173f87
+fn todo_throws_error() {
+    assert_js!(
+        r#"
+fn go() {
+    todo
+}
+"#,
+        r#""use strict";
+
+function go() {
+  throw Object.assign(
+    new Error("This has not yet been implemented"),
+    { gleam_error: "todo", module: "the_app", function: "go", line: 15 }
```

Comment:
Not bothered with any long names in the formatting test as this uses `wrap_args` and `wrap_object` that are already well tested for splitting over multiple lines properly.

Reply:
Makes sense, ta